test/unit/org/apache/cassandra/index/sai/cql/types/IndexingTypeSupport.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.cql.types;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.junit.Before;
+
+import org.apache.cassandra.index.sai.SAITester;
+
+public abstract class IndexingTypeSupport extends SAITester
+{
+    public static final int NUMBER_OF_VALUES = 64;
+
+    protected final DataSet<?> dataset;
+
+    private final boolean widePartitions;
+    private final Scenario scenario;
+    private Object[][] allRows;
+
+    public enum Scenario
+    {
+        MEMTABLE_QUERY
+    }
+
+    protected static Collection<Object[]> generateParameters(DataSet<?> dataset)
+    {
+        return Arrays.asList(new Object[][]
+        {
+            { dataset, true, Scenario.MEMTABLE_QUERY },
+            { dataset, false, Scenario.MEMTABLE_QUERY },
+        });
+    }
+
+    public IndexingTypeSupport(DataSet<?> dataset, boolean widePartitions, Scenario scenario)
+    {
+        this.dataset = dataset;
+        this.widePartitions = widePartitions;
+        this.scenario = scenario;
+    }
+
+    @Before
+    public void createTable()
+    {
+        dataset.init();
+
+        createTable(String.format("CREATE TABLE %%s (pk int, ck int, value %s, PRIMARY KEY(pk, ck))", dataset));
+
+        disableCompaction();
+
+        allRows = generateRows(dataset, widePartitions);
+    }
+
+    protected void runIndexQueryScenarios() throws Throwable
```

Comment:
If we make this method public and mark it with `@Test` we can save us the 101 repetitions of this block on the subclasses of `IndexingTypeSupport`:
```
@Test
public void test() throws Throwable
{
    runIndexQueryScenarios();
}
```
If that works, it would be 101 methods less to maintain :)

Reply:
I completely agree but I didn't enjoy getting rid of the 101 methods.

Reply:
Yes, sorry, it should have been a tedious task